common/djangoapps/course_modes/views.py

```diff
@@ -88,12 +88,15 @@ def get(self, request, course_id, error=None):
         # If there are both modes, default to non-id-professional.
         has_enrolled_professional = (CourseMode.is_professional_slug(enrollment_mode) and is_active)
         if CourseMode.has_professional_mode(modes) and not has_enrolled_professional:
-            redirect_url = reverse('verify_student_start_flow', kwargs={'course_id': unicode(course_key)})
+            purchase_workflow = request.GET.get("purchase_workflow", "single")
+            verify_url = reverse('verify_student_start_flow', kwargs={'course_id': unicode(course_key)})
+            redirect_url = "{url}?purchase_workflow={workflow}".format(url=verify_url, workflow=purchase_workflow)
             if ecommerce_service.is_enabled(request.user):
                 professional_mode = modes.get(CourseMode.NO_ID_PROFESSIONAL_MODE) or modes.get(CourseMode.PROFESSIONAL)
-                if professional_mode.sku:
+                if purchase_workflow == "single" and professional_mode.sku:
                     redirect_url = ecommerce_service.checkout_page_url(professional_mode.sku)
-
+                if purchase_workflow == "bulk" and professional_mode.bulk_sku:
```

Comment:
I debated doing an elif here.  I don't see a scenario where the second condition could override the first -- from a pure technical perspective I suppose we might save one CPU cycle by moving to elif and avoiding the second check in the case of a successful first check, but I actually prefer the clarity of the separate checks.